lib/dry/schema/message/or/multi_path.rb

```diff
@@ -14,23 +12,85 @@ module Or
         # @api public
         class MultiPath < Abstract
           # @api private
-          attr_reader :root
+          class MessageArray
+            # @api private
+            def initialize(messages)
+              @messages = messages.flatten
+            end
+
+            # @api private
+            def _paths
+              @messages.map(&:_path)
+            end
+
+            # @api private
+            def to_or(root)
+              self.class.new(@messages.map { _1.to_or(root) })
+            end
+
+            # @api private
+            def to_h
+              MessageSet.new(@messages).to_h
+            end
+          end
+
+          # @api private
+          def self.handler(message)
+            handlers.find { |k,| message.is_a?(k) }&.last
```

Comment:
specifically iterating over the hash here to get correct `is_a?` behavior, otherwise the hash lookup is a literal equality on the type